### PR TITLE
Added capability to skip DB healthcheck if blacklisting store is available

### PR DESCRIPTION
**Pull request by r0goyal**

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java [hunk 1]
```diff
@@ -117,7 +115,7 @@ private void init(final ImmutableList<Class<?>> inEntities) {
         val blacklistingStore = getBlacklistingStore();
         this.shardManager = createShardManager(numShards, blacklistingStore);
         this.shardInfoProvider = new ShardInfoProvider(dbNamespace);
-        this.healthCheckManager = new HealthCheckManager(dbNamespace, shardManager, shardInfoProvider);
+        this.healthCheckManager = new HealthCheckManager(dbNamespace, shardInfoProvider, blacklistingStore, shardManager);
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java]: Move these to constructor and make the properties final if possible

File: src/main/java/io/appform/dropwizard/sharding/sharding/InMemoryLocalShardBlacklistingStore.java [hunk 2]
```diff
@@ -41,4 +41,9 @@ public void unblacklist(int shardId) {
     public boolean blacklisted(int shardId) {
         return blacklisted.getOrDefault(shardId, false);
     }
+
+    @Override
+    public boolean disableNativeHealthCheck() {
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/sharding/InMemoryLocalShardBlacklistingStore.java]: Make this a config variable in `ShardedHibernateFactory`

File: src/main/java/io/appform/dropwizard/sharding/healthcheck/BlacklistingAwareHealthCheck.java [hunk 3]
```diff
@@ -24,9 +31,14 @@ protected Result check() {
             log.info("returning healthy since shard is blacklisted [{}]", shardId);
             return Result.healthy();
         }
+
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/healthcheck/BlacklistingAwareHealthCheck.java]: This should be the first thing to be checked

File: src/main/java/io/appform/dropwizard/sharding/sharding/ShardBlacklistingStore.java [hunk 4]
```diff
@@ -21,7 +21,13 @@
...
  */
 public interface ShardBlacklistingStore {
+
     void blacklist(int shardId);
+
     void unblacklist(int shardId);
+
     boolean blacklisted(int shardId);
+
+    boolean disableNativeHealthCheck();
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/sharding/ShardBlacklistingStore.java]: Remove

File: src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java (hunk 1 above)
Comment (r0goyal) [reply src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java]: This is part of init method called by each constructor. One of the constructor has few preprocessing steps which eliminates constructor chaining

File: src/main/java/io/appform/dropwizard/sharding/healthcheck/BlacklistingAwareHealthCheck.java (hunk 3 above)
Comment (r0goyal) [reply src/main/java/io/appform/dropwizard/sharding/healthcheck/BlacklistingAwareHealthCheck.java]: Done

File: src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java (hunk 1 above)
Comment (r0goyal) [reply src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java]: Done

**Pull request merged**